src/core_plugins/region_map/public/region_map_visualization.js

```diff
@@ -0,0 +1,138 @@
+import 'plugins/kbn_vislib_vis_types/controls/vislib_basic_options';
+import _ from 'lodash';
+import AggConfigResult from 'ui/vis/agg_config_result';
+import { BaseMapsVisualizationProvider } from '../../tile_map/public/base_maps_visualization';
+import ChoroplethLayer from './choropleth_layer';
+import { truncatedColorMaps }  from 'ui/vislib/components/color/truncated_colormaps';
+import AggResponsePointSeriesTooltipFormatterProvider from './tooltip_formatter';
+import 'ui/vis/map/service_settings';
+
+export function RegionMapsVisualizationProvider(Private, Notifier, config) {
+
+  const tooltipFormatter = Private(AggResponsePointSeriesTooltipFormatterProvider);
+  const BaseMapsVisualization = Private(BaseMapsVisualizationProvider);
+
+  return class RegionMapsVisualization extends BaseMapsVisualization {
+
+    constructor(container, vis) {
+      super(container, vis);
+      this._vis = this.vis;
+      this._choroplethLayer = null;
+      this._notify = new Notifier({ location: 'Region map' });
+    }
+
+
+    async render(esReponse, status) {
+      await super.render(esReponse, status);
+
+      if (!this._choroplethLayer) {
+        return;
+      }
+
+      if (this._choroplethLayer.isDataLoaded()) {
+        return;
+      }
+
+      return new Promise((resolve) => {
+        this._choroplethLayer.once('dataLoaded', () => {
+          resolve();
```

Comment:
Since the BaseMapsVisualization is triggering the `renderComplete` event, this could potentially happen before the 'dataLoaded' event.

Reply:
I don't think so. RegionMapVisualization does not fire an event. RegionMapsVisualization#render waits until BaseMapVisualization#render is complete. Only  after the main rendering is complete, we check whether the choroplethLayer has finished rendering. If not, we wait until it is completed.

Reply:
> RegionMapVisualization does not fire an event

That was the point I was trying to make, don't we want the `this._choroplethLayer` "dataLoaded" event to fire before we fire `renderComplete`?